File: 118. Pascal's Triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result = {{1}};

        for(int i=1;i<numRows;i++){
            vector<int> temp;
            for(int j=0;j<=i;j++){
                if(j == 0 || j == i) {
                    temp.push_back(1);
                    continue;
                }

                temp.push_back(result[i-1][j-1] + result[i-1][j]);
            }

            result.push_back(temp);
        }

        return result;
    }
};

File: 118. Pascal's Triangle/Solution.java
public class solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();
        
        // First row is always [1]
        result.add(new ArrayList<>());
        result.get(0).add(1);
        
        for (int i = 1; i < numRows; i++) {
            List<Integer> temp = new ArrayList<>();
            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    temp.add(1);
                } else {
                    int val = result.get(i - 1).get(j - 1) + result.get(i - 1).get(j);
                    temp.add(val);
                }
            }
            result.add(temp);
        }
    }
}
